Add the `Share Tests` section into the guide

mcasquer opened the pull request:
Partially covers #1721

Pull Request Checklist

* [x] write the documentation
* [x] include a release note

mcasquer commented on docs/examples.rst:
```diff
@@ -446,7 +446,28 @@ location:
+
+    # Discover tests located within this project's own repository
+    - name: project-specific
+        how: fmf
```
@happz updated !

mcasquer replied:
Thanks for pointing it out, updated !

bajertom commented on docs/examples.rst:
```diff
@@ -446,7 +446,28 @@ location:
+testing workflow.
+
+``tmt`` allows references to external repositories directly within the ``discover``
+step of the ``plan``. By specifying the url of a remote ``Git repository``, tmt
```
I wouldn't drop the double tick, but maybe lowered capital G in `Git repository` to keep it consistent with the rest of the documentation

psss reviewed (approved):
Thanks for preparing this. Looks good. We're just trying to slowly but surely get rid of the examples page and move the content to the guide where it will be nicely structured with enough context and comments. I've moved the content in 7e3f9515 and added a short release note about the new section.

mcasquer commented:
> Thanks for preparing this. Looks good. We're just trying to slowly but surely get rid of the examples page and move the content to the guide where it will be nicely structured with enough context and comments. I've moved the content in [7e3f951](https://github.com/teemtee/tmt/commit/7e3f9515f6ad86089b4b385692d0a5274ca27bd8) and added a short release note about the new section.

@psss thanks a lot !
